src/k8s-configuration/azext_k8s_configuration/_params.py

```diff
@@ -3,44 +3,134 @@
+    validate_operator_namespace
+)
 
-from azure.cli.core.commands.validators import get_default_location_from_resource_group
-from ._validators import _validate_configuration_type, _validate_operator_namespace, _validate_operator_instance_name
+from .action import (
+    KustomizationAddAction,
+)
+from . import consts
 
 
 def load_arguments(self, _):
-    sourcecontrolconfiguration_type = CLIArgumentType(help='Name of the Kubernetes Configuration')
-
     with self.argument_context('k8s-configuration') as c:
         c.argument('tags', tags_type)
-        c.argument('location',
-                   validator=get_default_location_from_resource_group)
-        c.argument('name', sourcecontrolconfiguration_type,
-                   options_list=['--name', '-n'])
+        c.argument('location', validator=get_default_location_from_resource_group)
         c.argument('cluster_name',
                    options_list=['--cluster-name', '-c'],
                    help='Name of the Kubernetes cluster')
         c.argument('cluster_type',
+                   options_list=['--cluster-type', '-t'],
                    arg_type=get_enum_type(['connectedClusters', 'managedClusters']),
-                   help='Specify Arc clusters or AKS managed clusters.')
+                   help='Specify Arc connected clusters or AKS managed clusters.')
+
+    with self.argument_context('k8s-configuration flux') as c:
+        c.argument('name',
+                   options_list=['--name', '-n'],
+                   help='Name of the flux configuration',
+                   validator=validate_fluxconfig_name)
+        c.argument('scope',
+                   options_list=['--scope', '-s'],
+                   arg_type=get_enum_type(['namespace', 'cluster']),
+                   help="Specify scope of the operator to be 'namespace' or 'cluster'")
+        c.argument('namespace',
+                   help='Namespace to deploy the configuration',
+                   options_list=['--namespace', '--ns'],
+                   validator=validate_namespace)
+        c.argument('kind',
+                   arg_type=get_enum_type([consts.GIT]),
+                   help='Source kind to reconcile')
+        c.argument('url',
+                   options_list=['--url', '-u'],
+                   help='URL of the git repo source to reconcile')
+        c.argument('timeout',
+                   help='Maximum time to reconcile the source before timing out')
+        c.argument('sync_interval',
+                   options_list=['--interval', '--sync-interval'],
+                   help='Time between reconciliations of the source on the cluster')
+        c.argument('branch',
+                   arg_group="Git Repo Ref",
+                   help='Branch within the git source to reconcile with the cluster')
+        c.argument('tag',
+                   arg_group="Git Repo Ref",
+                   help='Tag within the git source to reconcile with the cluster')
```

Comment:
May I ask is this the tag stored in ARM service? If so, do we need to consider using `tags_type`?

Reply:
This is not an ARM tag, this is a different thing. This is referring to a tag in the Git repository